### PR TITLE
Added hoverColor property on ExpansionTile widget

### LINKED ISSUE #94397: Add "hoverColor" property to CheckBoxListTile and ExpansionTile Widgets
## Use case

"hoverColor" property is implemented for a number of Widgets, but for some reason it has not yet been done for CheckBoxListTile and ExpansionTile which affects Web / Desktop UX.

Interesting, that "hoverColor" already exists in CheckBox Widget.

## Proposal

It's quite easy to add this property by modifying respected Widgets.

For CheckBoxListTile:
add to packages/flutter/lib/src/material/checkbox_list_tile.dart
```dart
  const CheckboxListTile({
    Key? key,
    required this.value,
    required this.onChanged,
    this.activeColor,
    this.checkColor,
    this.tileColor,
    this.title,
    this.subtitle,
    this.isThreeLine = false,
    this.dense,
    this.secondary,
    this.selected = false,
    this.controlAffinity = ListTileControlAffinity.platform,
    this.autofocus = false,
    this.contentPadding,
    this.tristate = false,
    this.shape,
    **this.hoverColor,**
    this.selectedTileColor,
  }) 
```

add property

  /// The color for the tile's [Material] when a pointer is hovering over it.
  **final Color? hoverColor;**       

and add into return method

    return MergeSemantics(
      child: ListTileTheme.merge(
        selectedColor: activeColor ?? Theme.of(context).toggleableActiveColor,
        child: ListTile(
          leading: leading,
          title: title,
          subtitle: subtitle,
          trailing: trailing,
          isThreeLine: isThreeLine,
          dense: dense,
          enabled: onChanged != null,
          onTap: onChanged != null ? _handleValueChange : null,
          selected: selected,
          autofocus: autofocus,
          contentPadding: contentPadding,
          shape: shape,
          selectedTileColor: selectedTileColor,
          tileColor: tileColor,
          **hoverColor: hoverColor,**
        ),
      ),
    ); 

ExpansionTile can be extended in the same way

I have created a pull request #91289 for these modification, but, unfortunately have not created test cases (have no experience) and the pull request was closed. Is it necessary to create new tests for such a minor update?.. If no, probably, the PR could be re-opened and merged.

Thanks!

### PR DESCRIPTION
**Pull request by M97Chahboun**
Fixes #94397

Adds a `hoverColor` property to `ExpansionTile` widget,

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by HansMuller** [comment]
We should use a `MaterialStateProperty<Color?>` valued overlayColor property the way Checkbox dxoes. See https://github.com/flutter/flutter/issues/94397#issuecomment-1335966249.

**Comment by M97Chahboun** [comment]
@HansMuller Please can you help me to close this PR and #116026
Thank you.

**Comment by Hixie** [comment]
as with https://github.com/flutter/flutter/pull/116026 it looks like some tests failed the last time this was updated, if you resolve the conflicts and repush it'll rerun the tests and we can see if they are real problems or not. Thanks!

**Comment by M97Chahboun** [comment]
@Hixie Thank you
I re-push please check it when you have time.

**Comment by Hixie** [comment]
Looks like the tests still failed (you should be able to see them in the GitHub UI here).

**Comment by Hixie** [comment]
(sorry, hit wrong button)

**Comment by Hixie** [comment]
@M97Chahboun Thanks for your contributions so far. Is this something you are still interested in working on?

**Comment by M97Chahboun** [comment]
Yes tomorrow I will try to solve tests issue

**Review by Piinks — COMMENTED** [review]
Hi @M97Chahboun, I just came across this. Apologies, it looks like it fell through the review queue cracks. Are you still interested in pursuing this change? If so, it should be rebased with the tip of the tree to update it, and then I think there was some feedback from @HansMuller above to apply.
Thank you for contributing!

**Comment by M97Chahboun** [comment]
OK I will try again

File: packages/flutter/test/material/expansion_tile_test.dart [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
 
+import 'dart:ui';
+
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:leak_tracker_flutter_testing/leak_tracker_flutter_testing.dart';
 
+import '../rendering/mock_canvas.dart';
+
```
Comment (Piinks) [inline packages/flutter/test/material/expansion_tile_test.dart:13]:
```suggestion
```

This should fix the failures here

File: packages/flutter/lib/src/material/expansion_tile.dart [hunk 2]
```diff
@@ -489,6 +490,9 @@ class ExpansionTile extends StatefulWidget {
   /// which means that the expansion arrow icon will appear on the tile's trailing edge.
   final ListTileControlAffinity? controlAffinity;
 
+  /// {@macro flutter.material.inkwell.hoverColor}
+  final Color? hoverColor;
```
Comment (Piinks) [inline packages/flutter/lib/src/material/expansion_tile.dart:494]: Should this be a MaterialStateProperty?

Comment (M97Chahboun) [reply packages/flutter/lib/src/material/expansion_tile.dart:494]: I suggest to use MaterialStateProperty on background color for handle all MaterialState colors...
and remove hover color

Comment (M97Chahboun) [reply packages/flutter/lib/src/material/expansion_tile.dart:494]: Screenshot 2023-12-12 at 11 25 44


Screenshot 2023-12-12 at 11 23 29

I think that he means ?

Comment (Piinks) [reply packages/flutter/lib/src/material/expansion_tile.dart:494]: Yep. I think so too! :)

Comment (Piinks) [reply packages/flutter/lib/src/material/expansion_tile.dart:494]: @M97Chahboun would you like to continue working on this?

Comment (M97Chahboun) [reply packages/flutter/lib/src/material/expansion_tile.dart:494]: Yes, but I have some difficulties on tests...

**Comment by goderbauer** [comment]
(triage) @M97Chahboun Do you still have plans to come back to this in the near future and address the feedback given above?

**Comment by M97Chahboun** [comment]
Yes, I have some issues on tests I will fix it ASAP

**Comment by Piinks** [comment]
Hi @M97Chahboun, thanks for the contribution! It looks like there are some failing tests and merge conflicts that need to be resolved here, can you take a look?

**Comment by Piinks** [comment]
Hey @M97Chahboun, just checking in. :) I saw some new commits, but it still looks like this change is failing. Are you still working on this?

**Comment by M97Chahboun** [comment]
Yes but i have some difficult with tests

**Comment by Piinks** [comment]
This wiki page has guidance on writing and running tests, will you let me know if that helps? https://github.com/flutter/flutter/wiki/Running-and-writing-tests

**Comment by Piinks** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to re-open or submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**